gen/dcos-config.yaml

```diff
@@ -8,7 +8,12 @@ package:
     content: |
       {
         "environment": {
+{% switch enable_mesos_ipv6_discovery %}
+{% case "false" %}
```

Comment:
The `MESOS_IP_DISCOVERY_COMMAND` should be independent of the `MESOS_IPV6_DISCOVERY_COMMAND`, i.e. even if the `MESOS_IPV6_DISCOVERY_COMMAND` is set the `MESOS_IP_DISCOVERY_COMMAND` should still be required. Mesos agents only listen to IPv4 addresses at this point. They take in an IPv6 address, but they never actually open an IPv6 socket. So we always need to specify and IPv4 address for the agent.

Reply:
I don't think that's true. If I specify MESOS_IPV6_DISCOVERY_COMMAND without a flag then cluster doesn't come up if its IPV4 only

Reply:
Did you check the cluster for the error you are seeing when you are specifying both these flags? I think you are specifying this flag for the master as well? The `MESOS_IPV6_DISCOVERY_COMMAND` flag exists only for the agent. So my guess is that if you specify it both for the master and the agent, the master might not be coming up.

Reply:
Just to confirm this the output from my local build:
```
[vagrant@centos8 bin]$ ./mesos-master.sh --help | grep discovery
                                                                                 with `--ip_discovery_command`.
  --ip_discovery_command=VALUE                                                   Optional IP discovery binary: if set, it is expected to emit
[vagrant@centos8 bin]$ ./mesos-agent.sh --help | grep discovery
  --appc_simple_discovery_uri_prefix=VALUE                         URI prefix to be used for simple discovery of appc images,
                                                                   with `--ip_discovery_command`.
                                                                   with '--ip6_discovery_command'.
  --ip6_discovery_command=VALUE                                    Optional IPv6 discovery binary: if set, it is expected to emit
  --ip_discovery_command=VALUE                                     Optional IP discovery binary: if set, it is expected to emit
[vagrant@centos8 bin]$
```
You can see that only the agent has the `ip6_discovery_command` flag.